src/lib/statistics/statistics_objects/min_max_filter.cpp

```diff
@@ -127,17 +127,17 @@ bool MinMaxFilter<T>::does_not_contain(const PredicateCondition predicate_condit
     case PredicateCondition::BetweenLowerExclusive: {
       Assert(static_cast<bool>(variant_value2), "Between operator needs two values.");
       const auto value2 = boost::get<T>(*variant_value2);
-      return value > max || value2 <= min;
+      return value >= max || value2 < min;
     }
     case PredicateCondition::BetweenUpperExclusive: {
       Assert(static_cast<bool>(variant_value2), "Between operator needs two values.");
       const auto value2 = boost::get<T>(*variant_value2);
-      return value >= max || value2 < min;
+      return value > max || value2 <= min;
```

Comment:
Help me here. `BetweenUpperExclusive` would mean a search interval as `[5, 10)`.
If my minimum is 5, I don't see how this change makes sense.
Same for `BetweenLowerExclusive`.

Reply:
Took me some time to wrap my head around it, but didn't find a better way to explain it.

Which side of the disjunction are you asking about? As you mention min, probably the right side. That uses value2, not value. In your example 10 is not smaller or equal to 5, so we cannot prune. If value2 was 5, e.g., [2, 5), the 5 would be exclusive, so we could prune.

Reply:
Added two examples, hopefully those help.

Reply:
Shouldn't review late at night. I think it makes perfect sense. Will take another look on the comments/tests next week.